components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectSystemClaimImpl.java

```diff
@@ -145,8 +145,11 @@ private boolean isCodeHashApplicable(String responseType) {
 
     private boolean isAccessTokenHashApplicable(String responseType) {
         // At_hash is generated on an access token. Therefore check whether the response type returns an access_token.
-        // id_token and none response types don't return and access token
-        return !OAuthConstants.ID_TOKEN.equalsIgnoreCase(responseType) &&
-                !OAuthConstants.NONE.equalsIgnoreCase(responseType);
+        for (String respType : responseType.split(" ")) {
```

Comment:
can we create a new private method to get an array of all response types? If we have a separate method, it can be used for other flows as well.

WDYT?

Reply:
Added a new method.